backend/src/v2/driver/driver.go

```diff
@@ -940,3 +1021,203 @@ func generateOutputURI(root, artifactName string, taskName string) string {
+	for key, val := range pvcAnnotationsInput.GetStructValue().AsMap() {
+		typedVal := val.(structpb.Value)
+		pvcAnnotations[key] = typedVal.GetStringValue()
+	}
+
+	// Optional input: volume_name
+	volumeNameInput := inputs.ParameterValues["volume_name"]
+	volumeName := volumeNameInput.GetStringValue()
+
+	// Initialize Kubernetes client set
+	restConfig, err := rest.InClusterConfig()
+	if err != nil {
+		return "", fmt.Errorf("failed to initialize kubernetes client: cannot get incluster config: %w", err)
+	}
+	k8sClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return "", fmt.Errorf("failed to initialize kubernetes client: %w", err)
+	}
+
+	// Create a PersistentVolumeClaim object
+	pvc := &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        pvcName,
+			Annotations: pvcAnnotations,
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes: accessModes,
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: resource.MustParse(volumeSizeInput.GetStringValue()),
+				},
+			},
+			StorageClassName: &storageClassName,
+			VolumeName:       volumeName,
+		},
+	}
+
+	// Create the PVC in the cluster
+	createdPVC, err := k8sClient.CoreV1().PersistentVolumeClaims(namespace).Create(context.Background(), pvc, metav1.CreateOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to create pvc: %w", err)
+	}
+	glog.Infof("Created PVC %s\n", createdPVC.ObjectMeta.Name)
+	return createdPVC.ObjectMeta.Name, nil
+}
+
+func DeletePVC(inputs *pipelinespec.ExecutorInput_Inputs, namespace string) error {
+	// Required input: pvc_name
+	pvcNameInput, ok := inputs.ParameterValues["pvc_name"]
+	if !ok {
+		return fmt.Errorf("failed to delete pvc: required parameter pvc_name not provided")
+	}
+	pvcName := pvcNameInput.GetStringValue()
+
+	// Initialize Kubernetes client set
+	restConfig, err := rest.InClusterConfig()
+	if err != nil {
+		return fmt.Errorf("failed to initialize kubernetes client: %w", err)
+	}
+	k8sClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return fmt.Errorf("failed to initialize kubernetes client set: %w", err)
+	}
+
+	// Get the PVC you want to delete, verify that it exists.
+	_, err = k8sClient.CoreV1().PersistentVolumeClaims(namespace).Get(context.TODO(), pvcName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to delete pvc %s: cannot find pvc: %v", pvcName, err)
+	}
+
+	// Delete the PVC.
+	err = k8sClient.CoreV1().PersistentVolumeClaims(namespace).Delete(context.TODO(), pvcName, metav1.DeleteOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to delete pvc %s: %v", pvcName, err)
+	}
+
+	glog.Infof("Deleted PVC %s\n", pvcName)
+	return nil
+}
+
+func makeVolumeMountPatch(pvcMount []*kubernetesplatform.PvcMount, task *pipelinespec.PipelineTaskSpec, dag *metadata.DAG, dagTasks map[string]*metadata.Execution) ([]corev1.VolumeMount, []corev1.Volume, error) {
```

Comment:
NIT: add a comment describing the function

Reply:
Good point. I added some comments for this.